cli.js

```diff
@@ -2,25 +2,33 @@
 var Cabal = require('cabal-core')
 var swarm = require('cabal-core/swarm.js')
 var minimist = require('minimist')
+var os = require('os')
+var fs = require('fs')
+var yaml = require('js-yaml')
+var mkdirp = require('mkdirp')
 var frontend = require('./neat-screen.js')
 
 var args = minimist(process.argv.slice(2))
 
-var homedir = process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE
-var rootdir = args.dir || (homedir + '/.cabal/archives/')
+var protocolMajorVersion = Cabal.protocolVersion.split('.')[0]
```

Comment:
no, that only happens if we make a breaking change (major version). and if we make a breaking change, i think we should be free to rid ourselves of the headache of old configs and make the best of the breaking change.

this is definitely to try to pin down our issues. we _should_ never need to do this, but _if_ we do it then we can exclude for certain that old clients are screwing things up.

i am fine with trying out this approach for a while (after this pr is stable and merged) and then reverting to not using a protocol version if it still feels really bad

Reply:
I think we're saying the same things:

- I agree with this change (using a new dir to escape the vestigial hypercore issue)
- I don't like it being tied to *protocol version* though, since this is more like a *database version* bump
- If we conflate protocol and db versioning, then every time we do a major protocol change we'll be essentially deleting (hiding) users' private keys needlessly by creating a new cabal directory